### ISSUE TITLE
Rewrite frontend

### ISSUE THREAD

**Issue by afruth** [jshacks/community2.jshacks.io#4]
The frontend was hacked together in a very short time, so it's not the best, structure-wise.

We should rebuild it from scratch either using Blaze like now and adding routes or moving to React and adding routes.

@dumconstantin what do you say? I would be comfortable with Blaze.

@IustiNN are you in as a fellow Meteorite? :)

**Comment by iustin-nita** [comment]
Sure thing :)

**Comment by afruth** [comment]
K, then, let's proceed.

We need to figure out an initial route structure, i'll mockup something sometime tonight. 

Also we'll branch and PR, no more master nonsense, hacking has ended, let the programming begin :D

**Comment by afruth** [comment]
Mai dureaza un pic, sa lamuresc cu @dumconstantin in ce directie o va lua dezvoltarea.

**Comment by dumconstantin** [comment]
Hi,

Here's are some general pointers for making community.jshacks.io:

**Tagline**: Where the community starts hacking

**Objective**: Make the efforts & achievements of members visible and encourage & streamline the process of contributing for new members

**Mission**: Jump-start the open source movement in Romania 

**Technology Considerations**:
(please take this part as open to debate) 

The architecture is rather straightforward:
-- a job system that aggregates information from Github and stores it in a structured way
-- a client that provides visualization of that data without complex user interaction

In this scenario an end to end framework will make the two systems coupled which will increase development cycles, decrease contributors potential and have a tool on a more fragile infrastructure to maintain.

To overcome this, we can opt for a real time database that acts as glue between the job system and the visualization part (e.g. Firebase or similar) 

``` code
Job --> |~~~~~~~~~~~~~~|
Job --> | Real Time DB | --> Client
Job --> |~~~~~~~~~~~~~~|
```

**Job System**
Jobs are single files that use the **octonode** module to connect to github, fetch data and push data to the RealTime DB.

This decreases our costs for a DO instance - it will only run jobs at a certain interval (not necessary in parallel). As a V2 we could migrate these Jobs in a lambda system and have a scheduling & cost per processing out of the box.

**Client system**
Firebase has CDN delivery so the Client code can be hosted there without the need for a server to deliver the client assets, app.js and index.html files.

When the client instantiates it creates a link to the Firebase instance and will be updated on every change.

We can use any Front End system (React/Angular/etc) but the lighter the better - I'm proposing RiotJS as it has a very small learning curve. You can check out and example here (a simple TodoMVC):
https://github.com/riot/examples/blob/gh-pages/todo-app/todo.tag

---

Developing these systems can be done in parallel by different teams without one knowing what the other is doing.

The only thing that matters is that the data structure is consistent in the Real Time DB.

---

**General Structure**:
- community.jshacks.io/projects/PROJECT-NAME
  -- Name, Description
  -- Authors (personal links, pics, etc)
  -- PowerPoint / Video Presentation
  -- Technologies (logos + lines of code)
  -- Stats (lines of code, commits, stars, watchers, forks, etc)
  -- Link to "View Online"
- community.jshacks.io/members/MEMBER-NAME
  -- Name, Description
  -- Personal Links (linkedin, blog, etc)
  -- Stats (lines of code, commits, followers, following, etc)
  -- Contributor on Projects
- Activity/History
- community.jshacks.io/communities/COMMUNITY-NAME
  -- Name, Description
  -- Past and Current Events (integrate with Meetup / Facebook)
  -- Members List (correlated with github users)
  -- Saved presentations from Speakers
- community.jshacks.io
  -- Organization Graphic (commits over time/contributions over time)
  -- Stats (locs, star, no.members, etc)
  -- Events (with stats)
  -- Communities (with stats)

of course add to these the connection pages with the lists of members, communities, projects, etc

**Comment by afruth** [comment]
I agree with the above. We can use Riot.js as a lightweight and easy to learn FrontEnd library.

I also think that we should build it from scratch, using a new repo.

**Comment by iustin-nita** [comment]
How about Vue.js?

**Comment by dumconstantin** [comment]
Vue is better that Riot in terms of performance and scalability and has a much bigger community behind it so over the long haul I'm sure we'll be using Vue.

In the meantime, Riot is the simple one out of the pack and is quite intuitive, we can use it to map out the entire platform quickly and later change it to Vue in a more methodical way. 

Here's how Riot will look like:

``` html
<todo>
    <h3>TODO</h3>

    <ul>
        <li each={ item, i in items }>{ item }</li>
    </ul>
    <form onsubmit={ handleSubmit }>
        <input>
        <button>Add #{ items.length + 1 }</button>
    </form>

    this.items = []

    handleSubmit(e) {
        var input = e.target[0]
        this.items.push(input.value)
        input.value = ''
    }
</todo>
```

What do you think?

**Comment by afruth** [comment]
I would be ok with both. We could go straight for Vue.js, we have now a placeholder site, built with Meteor and we have some time available to build the new one.

**Comment by dumconstantin** [comment]
Ok then, Vue it is, let's get hacking :)

I'll move this thread into a Wiki on [community.jshacks.io](https://github.com/jshacks/community.jshacks.io) where we can continue development.

See you guys there! :)
